app/src/test/java/org/schabi/newpipe/util/LocalizationTest.kt

```diff
@@ -0,0 +1,39 @@
+package org.schabi.newpipe.util
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.ocpsoft.prettytime.PrettyTime
+import java.text.SimpleDateFormat
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.util.GregorianCalendar
+import java.util.Locale
+
+class LocalizationTest {
+
+    @Test
+    fun `After initializing pretty time relativeTime() with a Calendar must work`() {
+        val reference = SimpleDateFormat("yyyy/MM/dd").parse("2021/1/1")
+        Localization.initPrettyTime(PrettyTime(reference, Locale.ENGLISH))
+
+        val actual = Localization.relativeTime(GregorianCalendar(2021, 1, 6))
+
+        assertEquals("1 month from now", actual)
```

Comment:
Maybe I am just unable to understand Kotlin, in this case please enlighten me. But this assertion will never be true because both times are just 5 days apart from each other - especially when I compare it with the last test? So shouldn't it be `assertEquals("5 days from now", actual)` here as well?

Reply:
You are absolutly correct, but this is actually what pretty time returns. I would guess it's either a bug in `PrettyTime` or a limitation when comparing with `GregorianCalendar`.

For these tests i went with the assumption that the current state is correct. So i just added assertions with what is returned, so regressions are covered.

Reply:
okay, can you add this explanation as an inline comment so that no one stumbles upon this later?

Reply:
Do you mean the first paragraph of the second (or both)?

I would add the first with the note that further research is required

Reply:
just add anything meaningful like `// yes this assertion is true, even if it should be 5 days, it works as it is. Future research required`

Maybe it has to do something with timezone shift. Could it be that Locale.ENGLISH is defaulting to the global standard locale US in Android? This would explain why 2021, 1, 1 is in fact 2020,12,12 normalized to UTC and could explain "1 month from now"